components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/provisioning/impl/DefaultProvisioningHandler.java

```diff
@@ -149,6 +155,20 @@ public void handle(List<String> roles, String subject, Map<String, String> attri
                  */
                 IdentityUtil.threadLocalProperties.get().put(FrameworkConstants.JIT_PROVISIONING_FLOW, true);
                 if (!userClaims.isEmpty()) {
+
+                    Claim[] claimList = userStoreManager.getUserClaimValues(
+                            UserCoreUtil.removeDomainFromName(username), null);
+                    List<Claim> existingUserClaimList = new ArrayList<>(Arrays.asList(claimList));
+                    Set<String> notToDeleteClaimSet = getNotToDeleteClaims();
```

Comment:
Shall we put a comment here, mentioning what we are going to achieve with this logic. It would be helpful in the future. WDYT?